Sources/PostgresNIO/New/Connection State Machine/ExtendedQueryStateMachine.swift

```diff
@@ -292,10 +330,36 @@ struct ExtendedQueryStateMachine {
                     return .succeedQuery(eventLoopPromise, with: result)
                 }
 
-            case .prepareStatement:
+            case .prepareStatement, .simpleQuery:
                 preconditionFailure("Invalid state: \(self.state)")
             }
-            
+
+        case .messagesSent(let context):
+            switch context.query {
+            case .simpleQuery(_, let eventLoopGroup):
+                return self.avoidingStateMachineCoW { state -> Action in
+                    state = .commandComplete(commandTag: commandTag)
+                    let result = QueryResult(value: .noRows(.tag(commandTag)), logger: context.logger)
+                    return .succeedQuery(eventLoopGroup, with: result)
+                }
+
+            case .unnamed, .executeStatement, .prepareStatement:
+                return self.setAndFireError(.unexpectedBackendMessage(.commandComplete(commandTag)))
+            }
+
+        case .rowDescriptionReceived(let context, _):
+            switch context.query {
+            case .simpleQuery(_, let eventLoopPromise):
+                return self.avoidingStateMachineCoW { state -> Action in
+                    state = .commandComplete(commandTag: commandTag)
+                    let result = QueryResult(value: .noRows(.tag(commandTag)), logger: context.logger)
+                    return .succeedQuery(eventLoopPromise, with: result)
+                }
```

Comment:
If a `CommandComplete` is received and state is `RowDescription`, It means we had a query that was in fact querying some rows, but it just happened that there were no rows matching the query.